winpr/libwinpr/file/generic.c

```diff
@@ -938,6 +938,7 @@ HANDLE FindFirstFileW(LPCWSTR lpFileName, LPWIN32_FIND_DATAW lpFindFileData)
 		}
 	}
 
+	free(fd);
 	return h;
 }
```

Comment:
It's a matter of style, but with functions that deal with resources I personally like to have a single return point and structure error handling with goto. This ensures that the resources are always freed.

Like this:
```diff
 	if (h != INVALID_HANDLE_VALUE)
 	{
 		if (!ConvertFindDataAToW(fd, lpFindFileData))
 		{
 			SetLastError(ERROR_NOT_ENOUGH_MEMORY);
-			free(fd);
-			return INVALID_HANDLE_VALUE;
+			h = INVALID_HANDLE_VALUE;
+			goto out;
  		}
  	}
-
+out:
+	free(fd);
  	return h;
  }
```
Though maybe it's an overkill in this case.